frontend/src/pages/Compare.tsx

```diff
@@ -14,334 +14,44 @@
-          const configs = await OutputArtifactLoader.load(
-            path,
-            workflowObjects[0]?.metadata?.namespace,
-          );
-          configs.forEach(config => {
-            const currentList: TaggedViewerConfig[] = viewersMap.get(config.type) || [];
-            currentList.push({
-              config,
-              runId: runs[i].run!.id!,
-              runName: runs[i].run!.name!,
-            });
-            viewersMap.set(config.type, currentList);
-          });
-        }
-      }),
-    );
-
-    // For each output artifact type, list all artifact instances in all runs
-    this.setStateSafe({ viewersMap });
-  }
-
-  protected _selectionChanged(selectedIds: string[]): void {
-    this.setState({ selectedIds });
-    this._loadParameters(selectedIds);
-    this._loadMetrics(selectedIds);
-  }
-
-  private _collapseAllSections(): void {
-    const collapseSections = {
-      [overviewSectionName]: true,
-      [paramsSectionName]: true,
-      [metricsSectionName]: true,
-    };
-    Array.from(this.state.viewersMap.keys()).forEach(t => {
-      const sectionName = componentMap[t].prototype.getDisplayName();
-      collapseSections[sectionName] = true;
-    });
-    this.setState({ collapseSections });
-  }
-
-  private _loadParameters(selectedIds: string[]): void {
-    const { runs, workflowObjects } = this.state;
-
-    const selectedIndices = selectedIds.map(id => runs.findIndex(r => r.run!.id === id));
-    const filteredRuns = runs.filter((_, i) => selectedIndices.indexOf(i) > -1);
-    const filteredWorkflows = workflowObjects.filter((_, i) => selectedIndices.indexOf(i) > -1);
-
-    const paramsCompareProps = CompareUtils.getParamsCompareProps(filteredRuns, filteredWorkflows);
-
-    this.setState({ paramsCompareProps });
-  }
-
-  private _loadMetrics(selectedIds: string[]): void {
-    const { runs } = this.state;
-
-    const selectedIndices = selectedIds.map(id => runs.findIndex(r => r.run!.id === id));
-    const filteredRuns = runs.filter((_, i) => selectedIndices.indexOf(i) > -1).map(r => r.run!);
-
-    const metricsCompareProps = CompareUtils.multiRunMetricsCompareProps(filteredRuns);
-
-    this.setState({ metricsCompareProps });
+import CompareV1, { CompareState } from './CompareV1';
+import CompareV2 from './CompareV2';
+import { PageProps } from './Page';
+
+export type CompareProps = PageProps & CompareState;
+
+// This is a router to determine whether to show V1 or V2 compare page.
+export default function Compare(props: CompareProps) {
+  const queryParamRunIds = new URLParser(props).get(QUERY_PARAMS.runlist);
+  const runIds = (queryParamRunIds && queryParamRunIds.split(',')) || [];
+
+  // Retrieves run details.
+  const { isSuccess, data } = useQuery<ApiRunDetail[], Error>(
+    ['run_details', { ids: runIds }],
+    () => Promise.all(runIds.map(async id => await Apis.runServiceApi.getRun(id))),
+    { staleTime: Infinity },
+  );
+
+  if (data === undefined) {
```

Comment:
What to do if there is an error? There is a `isError` data field in the return value of `useQuery`.

When there is error, here is an example of how to show an error banner: https://github.com/kubeflow/pipelines/blob/master/frontend/src/pages/Page.tsx#L70-L81

Reply:
You are right with your implementation! by using `updateBanner` directly! The `showPageError` is a highlight for example that I want to show you, but it doesn't mean that you should call `showPageError` directly.

Reply:
Ok great, thank you!